### ISSUE TITLE
Hidden Field IgnoresFormat is not working correctly

### ISSUE THREAD

**Issue by christopherpicard** [checkstyle/checkstyle#607]
Hidden Field IgnoreFormat is not working correctly. Here is some sample code:
```
public class Sample {
   private List<String> tests;
   public Sample() {
      tests = new ArrayList<>();
   }
   public List<String> getTests() {
      return tests;
   }
   public void addTest(final String test) {
      this.tests.add(test);
   }
   public void addAllTests(final List<String> tests) {
      this.tests.addAll(tests);
   }
}
```
Entry in Checkstyle xml:
```
     <module name="HiddenField">
      <property name="severity" value="warning"/>
      <property name="ignoreFormat" value="addAll.*"/>
      <property name="ignoreConstructorParameter" value="true"/>
      <property name="ignoreSetter" value="true"/>
    </module>
```
It is still flagging the addAll method as a violation.
I tried "addAll", "addAll*" as well just encase I was missing something with how the format works.
I am using the latest version of checkstyle 6.2

**Comment by romani** [comment]
yes ,it is a problem
~/checkstyle-test$ java -jar /home/rivanov/checkstyle-test/checkstyle/target/checkstyle-6.3-SNAPSHOT-all.jar     -c hiddenfield.xml  Test.java 
Starting audit...
/home/rivanov/checkstyle-test/Test.java:12:47: warning: 'tests' hides a field.
Audit done.

**Comment by alexkravin** [comment]
From docs:
http://checkstyle.sourceforge.net/config_coding.html#HiddenField
```
ignoreFormat	pattern for names to ignore
```

There's no mention about "Method name" - it's pattern to ignore <b>the name of variable or param</b>

So, just configure the Check as:
```
<module name="HiddenField">
      <property name="severity" value="warning"/>
      <property name="ignoreFormat" value="tests"/>
      <property name="ignoreConstructorParameter" value="true"/>
      <property name="ignoreSetter" value="true"/>
</module>
```

**Comment by christopherpicard** [comment]
If the intention of the ignoreFormat is to be filtering out parameter names, then maybe this needs to be written up as an enhancement. Specifying the parameter filter at least for my purposes is useless. I might want to have it in place on all methods but my versions of mutators and accessors. Also you would have to specify it for every variable in all classes which could change and just is to much overhead.

Either the ability of filtering method names should be added(prefered) or the ignoreSetters needs to be updated to include add/addAll methods to handle proper encapsulation of lists.

**Comment by christopherpicard** [comment]
I believe Issue #619 came to the same conclusion for a different issue that method name filtering should be provided for this check.

**Comment by alexkravin** [comment]
@christopherpicard,

ok, I agree it has sense, so I advise you to close this issue and open the new one, to not lose main sense in the depth of comments, with detailed specification about expectations from Check's updates, I'll proceed requested updates.

**Comment by romani** [comment]
@alexkravin, please update javadoc/xdoc for that field to clearly state what is a target of that Format. 

If users do mistakes, and even we do not catch that easily - smth should be fixed. 

In context of this problem - documentation should be updated.

**Comment by romani** [comment]
will be in 6.5

**Comment by mkordas** [comment]
released as 6.4.1
